### ISSUE TITLE
osmcoastline binary references build path

### ISSUE THREAD

**Issue by sebastic** [osmcode/osmcoastline#33]
The lintian QA tool reported that the `osmcoastline` binary references the build path, and this may cause reproducibility issues. To quote the tag description:
> The listed file or maintainer script appears to reference the build path used to build the package as specified in the Build-Path field of the .buildinfo file.
> 
> This is likely to cause the package to be unreproducible, but it may also indicate that the package will not work correctly outside of the maintainer's own system.

Grepping for the build path in the strings of the binary results in:
```
$ strings /tmp/osmcoastline_2.2.2-1_amd64/usr/bin/osmcoastline | grep "/build/osmcoastline-2.2.2"
/build/osmcoastline-2.2.2/src/osmcoastline.cpp
/build/osmcoastline-2.2.2/src/coastline_ring_collection.hpp
/build/osmcoastline-2.2.2/src/coastline_ring.hpp
/build/osmcoastline-2.2.2/src/coastline_ring.cpp
/build/osmcoastline-2.2.2/src/coastline_ring_collection.cpp
/build/osmcoastline-2.2.2/src/coastline_polygons.cpp
```

I wonder why these paths to the sources are contained in the binary, and how we can prevent them from being included.

**Comment by sebastic** [comment]
In the context of Debian packages `RelWithDebInfo` is preferred, so we can provide `dbgsym` packages with debug symbols automatically. The binaries are stripped as part of the package build as well.

Perhaps using relative paths for the `assert`s is a good compromise (e.g. only `src/osmcoastline.cpp`).

**Comment by joto** [comment]
I can't change the internals of the assert, so I can't put relative paths in there. But asserts shouldn't be in release builds anyway. I have changed the `RelWithDebInfo` build to disable asserts and this should fix the problem. Will do a release shortly.

**Comment by sebastic** [comment]
This issue is resolved by explicitly building (2.2.3) with `-DCMAKE_BUILD_TYPE=RelWithDebInfo` instead of `-DCMAKE_BUILD_TYPE=None` as debhelper sets by default.
